src/remote/utils.go

```diff
@@ -0,0 +1,88 @@
+package remote
+
+import (
+	"crypto/sha1"
+	"encoding/hex"
+	"fmt"
+	"os"
+	"time"
+
+	pb "github.com/bazelbuild/remote-apis/build/bazel/remote/execution/v2"
+	"github.com/bazelbuild/remote-apis/build/bazel/semver"
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/ptypes/timestamp"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// digestMessage calculates the digest of a proto message as described in the
+// Digest message's comments.
+func digestMessage(msg proto.Message) *pb.Digest {
+	digest, _ := digestMessageContents(msg)
+	return digest
+}
+
+// digestMessageContents is like DigestMessage but returns the serialised contents as well.
+func digestMessageContents(msg proto.Message) (*pb.Digest, []byte) {
+	b, err := proto.Marshal(msg)
+	if err != nil {
+		// Not really sure if there is a valid possibility to bring us here (given that
+		// the messages in question have no required fields) so assume it won't happen :)
+		log.Fatalf("Failed to marshal message: %s", err)
+	}
+	sum := sha1.Sum(b)
+	return &pb.Digest{
+		Hash:      hex.EncodeToString(sum[:]),
+		SizeBytes: int64(len(b)),
+	}, b
+}
+
+// lessThan returns true if the given semver instance is less than another one.
+func lessThan(a, b *semver.SemVer) bool {
```

Comment:
Somewhat surprised this doesn't exist in the SemVer package

Reply:
It's a proto package so doesn't have any helper methods like that (and they didn't add any ala the `ptypes` stuff). I guess maybe they don't want to maintain them for every language, although agreed this was rather less pleasant than using `coreos/semver` :(